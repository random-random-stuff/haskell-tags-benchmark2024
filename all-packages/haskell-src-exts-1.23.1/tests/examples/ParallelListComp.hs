{-# LANGUAGE ParallelListComp #-}

f xs ys zs = [ (x,y,z) | x <- xs | y <- ys, y > 2 | z <- zs ]
