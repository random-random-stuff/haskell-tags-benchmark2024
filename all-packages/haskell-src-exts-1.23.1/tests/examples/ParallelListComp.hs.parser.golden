ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan =
            SrcSpan "tests/examples/ParallelListComp.hs" 1 1 4 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/ParallelListComp.hs" 1 1 1 1
            , SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 1
            , SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 1
            , SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 1
            , SrcSpan "tests/examples/ParallelListComp.hs" 4 1 4 1
            , SrcSpan "tests/examples/ParallelListComp.hs" 4 1 4 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/ParallelListComp.hs" 1 1 1 34
            , srcInfoPoints =
                [ SrcSpan "tests/examples/ParallelListComp.hs" 1 1 1 13
                , SrcSpan "tests/examples/ParallelListComp.hs" 1 31 1 34
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/ParallelListComp.hs" 1 14 1 30
                , srcInfoPoints = []
                }
              "ParallelListComp"
          ]
      ]
      []
      [ FunBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 62
            , srcInfoPoints = []
            }
          [ Match
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 62
                , srcInfoPoints = []
                }
              (Ident
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/ParallelListComp.hs" 3 1 3 2
                   , srcInfoPoints = []
                   }
                 "f")
              [ PVar
                  SrcSpanInfo
                    { srcInfoSpan =
                        SrcSpan "tests/examples/ParallelListComp.hs" 3 3 3 5
                    , srcInfoPoints = []
                    }
                  (Ident
                     SrcSpanInfo
                       { srcInfoSpan =
                           SrcSpan "tests/examples/ParallelListComp.hs" 3 3 3 5
                       , srcInfoPoints = []
                       }
                     "xs")
              , PVar
                  SrcSpanInfo
                    { srcInfoSpan =
                        SrcSpan "tests/examples/ParallelListComp.hs" 3 6 3 8
                    , srcInfoPoints = []
                    }
                  (Ident
                     SrcSpanInfo
                       { srcInfoSpan =
                           SrcSpan "tests/examples/ParallelListComp.hs" 3 6 3 8
                       , srcInfoPoints = []
                       }
                     "ys")
              , PVar
                  SrcSpanInfo
                    { srcInfoSpan =
                        SrcSpan "tests/examples/ParallelListComp.hs" 3 9 3 11
                    , srcInfoPoints = []
                    }
                  (Ident
                     SrcSpanInfo
                       { srcInfoSpan =
                           SrcSpan "tests/examples/ParallelListComp.hs" 3 9 3 11
                       , srcInfoPoints = []
                       }
                     "zs")
              ]
              (UnGuardedRhs
                 SrcSpanInfo
                   { srcInfoSpan =
                       SrcSpan "tests/examples/ParallelListComp.hs" 3 12 3 62
                   , srcInfoPoints =
                       [ SrcSpan "tests/examples/ParallelListComp.hs" 3 12 3 13 ]
                   }
                 (ParComp
                    SrcSpanInfo
                      { srcInfoSpan =
                          SrcSpan "tests/examples/ParallelListComp.hs" 3 14 3 62
                      , srcInfoPoints =
                          [ SrcSpan "tests/examples/ParallelListComp.hs" 3 14 3 15
                          , SrcSpan "tests/examples/ParallelListComp.hs" 3 24 3 25
                          , SrcSpan "tests/examples/ParallelListComp.hs" 3 34 3 35
                          , SrcSpan "tests/examples/ParallelListComp.hs" 3 43 3 44
                          , SrcSpan "tests/examples/ParallelListComp.hs" 3 51 3 52
                          , SrcSpan "tests/examples/ParallelListComp.hs" 3 61 3 62
                          ]
                      }
                    (Tuple
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/ParallelListComp.hs" 3 16 3 23
                         , srcInfoPoints =
                             [ SrcSpan "tests/examples/ParallelListComp.hs" 3 16 3 17
                             , SrcSpan "tests/examples/ParallelListComp.hs" 3 18 3 19
                             , SrcSpan "tests/examples/ParallelListComp.hs" 3 20 3 21
                             , SrcSpan "tests/examples/ParallelListComp.hs" 3 22 3 23
                             ]
                         }
                       Boxed
                       [ Var
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/ParallelListComp.hs" 3 17 3 18
                             , srcInfoPoints = []
                             }
                           (UnQual
                              SrcSpanInfo
                                { srcInfoSpan =
                                    SrcSpan "tests/examples/ParallelListComp.hs" 3 17 3 18
                                , srcInfoPoints = []
                                }
                              (Ident
                                 SrcSpanInfo
                                   { srcInfoSpan =
                                       SrcSpan "tests/examples/ParallelListComp.hs" 3 17 3 18
                                   , srcInfoPoints = []
                                   }
                                 "x"))
                       , Var
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/ParallelListComp.hs" 3 19 3 20
                             , srcInfoPoints = []
                             }
                           (UnQual
                              SrcSpanInfo
                                { srcInfoSpan =
                                    SrcSpan "tests/examples/ParallelListComp.hs" 3 19 3 20
                                , srcInfoPoints = []
                                }
                              (Ident
                                 SrcSpanInfo
                                   { srcInfoSpan =
                                       SrcSpan "tests/examples/ParallelListComp.hs" 3 19 3 20
                                   , srcInfoPoints = []
                                   }
                                 "y"))
                       , Var
                           SrcSpanInfo
                             { srcInfoSpan =
                                 SrcSpan "tests/examples/ParallelListComp.hs" 3 21 3 22
                             , srcInfoPoints = []
                             }
                           (UnQual
                              SrcSpanInfo
                                { srcInfoSpan =
                                    SrcSpan "tests/examples/ParallelListComp.hs" 3 21 3 22
                                , srcInfoPoints = []
                                }
                              (Ident
                                 SrcSpanInfo
                                   { srcInfoSpan =
                                       SrcSpan "tests/examples/ParallelListComp.hs" 3 21 3 22
                                   , srcInfoPoints = []
                                   }
                                 "z"))
                       ])
                    [ [ QualStmt
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/ParallelListComp.hs" 3 26 3 33
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/ParallelListComp.hs" 3 28 3 30 ]
                            }
                          (Generator
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/ParallelListComp.hs" 3 26 3 33
                               , srcInfoPoints =
                                   [ SrcSpan "tests/examples/ParallelListComp.hs" 3 28 3 30 ]
                               }
                             (PVar
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 26 3 27
                                  , srcInfoPoints = []
                                  }
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 26 3 27
                                     , srcInfoPoints = []
                                     }
                                   "x"))
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 31 3 33
                                  , srcInfoPoints = []
                                  }
                                (UnQual
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 31 3 33
                                     , srcInfoPoints = []
                                     }
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 31 3 33
                                        , srcInfoPoints = []
                                        }
                                      "xs"))))
                      ]
                    , [ QualStmt
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/ParallelListComp.hs" 3 36 3 43
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/ParallelListComp.hs" 3 38 3 40 ]
                            }
                          (Generator
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/ParallelListComp.hs" 3 36 3 43
                               , srcInfoPoints =
                                   [ SrcSpan "tests/examples/ParallelListComp.hs" 3 38 3 40 ]
                               }
                             (PVar
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 36 3 37
                                  , srcInfoPoints = []
                                  }
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 36 3 37
                                     , srcInfoPoints = []
                                     }
                                   "y"))
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 41 3 43
                                  , srcInfoPoints = []
                                  }
                                (UnQual
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 41 3 43
                                     , srcInfoPoints = []
                                     }
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 41 3 43
                                        , srcInfoPoints = []
                                        }
                                      "ys"))))
                      , QualStmt
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/ParallelListComp.hs" 3 45 3 50
                            , srcInfoPoints = []
                            }
                          (Qualifier
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/ParallelListComp.hs" 3 45 3 50
                               , srcInfoPoints = []
                               }
                             (InfixApp
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 45 3 50
                                  , srcInfoPoints = []
                                  }
                                (Var
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 45 3 46
                                     , srcInfoPoints = []
                                     }
                                   (UnQual
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 45 3 46
                                        , srcInfoPoints = []
                                        }
                                      (Ident
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/ParallelListComp.hs" 3 45 3 46
                                           , srcInfoPoints = []
                                           }
                                         "y")))
                                (QVarOp
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 47 3 48
                                     , srcInfoPoints = []
                                     }
                                   (UnQual
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 47 3 48
                                        , srcInfoPoints = []
                                        }
                                      (Symbol
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/ParallelListComp.hs" 3 47 3 48
                                           , srcInfoPoints = []
                                           }
                                         ">")))
                                (Lit
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 49 3 50
                                     , srcInfoPoints = []
                                     }
                                   (Int
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 49 3 50
                                        , srcInfoPoints = []
                                        }
                                      2
                                      "2"))))
                      ]
                    , [ QualStmt
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/ParallelListComp.hs" 3 53 3 60
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/ParallelListComp.hs" 3 55 3 57 ]
                            }
                          (Generator
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/ParallelListComp.hs" 3 53 3 60
                               , srcInfoPoints =
                                   [ SrcSpan "tests/examples/ParallelListComp.hs" 3 55 3 57 ]
                               }
                             (PVar
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 53 3 54
                                  , srcInfoPoints = []
                                  }
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 53 3 54
                                     , srcInfoPoints = []
                                     }
                                   "z"))
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/ParallelListComp.hs" 3 58 3 60
                                  , srcInfoPoints = []
                                  }
                                (UnQual
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan "tests/examples/ParallelListComp.hs" 3 58 3 60
                                     , srcInfoPoints = []
                                     }
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/ParallelListComp.hs" 3 58 3 60
                                        , srcInfoPoints = []
                                        }
                                      "zs"))))
                      ]
                    ]))
              Nothing
          ]
      ]
  , []
  )
